### DISCUSSION TITLE
Support for Diffusers Custom Pipelines

### DISCUSSION THREAD

**Discussion by setothegreat** [vladmandic/sdnext#611] [Ideas]
I rarely see anybody bring this up, but [Diffusers has supported Custom Pipelines for quite some time now](https://github.com/huggingface/diffusers/issues/841). 

Specific custom pipelines such as Clip Guided Stable Diffusion allow for CLIP guidance during generation, which while using tons of VRAM, can produce far higher quality results, as well as Long Prompt Weighting Stable Diffusion, which removes the 75 token limit that can often plague more complex generations, seem as though they could be incredibly useful for those looking to take advantage of such features. [Features like this have been requested of Automatic1111 for quite a while now](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/735), but have thus far gone ignored.

[People have proven in the past that dedicated implementations of said pipelines are possible](https://github.com/space-nuko/stable-diffusion-webui/tree/feature/clip-guidance), but such repos are quite outdated, often dedicated to implementing only a single given pipeline, and are few and far between. The official thread in the first link makes it sound like a rather simple task, though I am not a programmer by any stretch, and so do not know specifically how such a thing could be implemented and passed into a WebUI interface, and then onto the actual Stable Diffusion process.

I am currently using WebUI UX as I find that repo it is a bit less prone to errors in it's current state, and because I do prefer how the UX is setup. However, if custom pipelines were implemented in this repo, not only would it easily become my dedicated repo for life, but if a Patreon were ever setup, I would become a donor for life.

**Comment by FullBleed** [comment]
> I rarely see anybody bring this up, but [Diffusers has supported Custom Pipelines for quite some time now](https://github.com/huggingface/diffusers/issues/841).
> 
> Specific custom pipelines such as Clip Guided Stable Diffusion allow for CLIP guidance during generation, which while using tons of VRAM, can produce far higher quality results, as well as Long Prompt Weighting Stable Diffusion, which removes the 75 token limit that can often plague more complex generations, seem as though they could be incredibly useful for those looking to take advantage of such features. [Features like this have been requested of Automatic1111 for quite a while now](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/735), but have thus far gone ignored.

@patrickvonplaten is currently developing diffuser support for this fork (#593 ), so I suspect he's the man to ping for this.

**Comment by BinaryQuantumSoul** [comment]
> I am currently using WebUI UX as I find that repo it is a bit less prone to errors in it's current state, and because I do prefer how the UX is setup. However, if custom pipelines were implemented in this repo, not only would it easily become my dedicated repo for life, but if a Patreon were ever setup, I would become a donor for life.

They are working on adding WebUIUX to Vlad themes btw
